cost/turbonomics/delete_unattached_volumes/aws/turbonomics_delete_virtual_volumes.pt

```diff
@@ -0,0 +1,364 @@
+
+#get turbonomic recommendation data
+datasource "ds_get_turbonomics_recommendations" do
+  request do
+    run_script $js_get_turbonomics_recommendations, $auth_cookie
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "vendorIDValues", jmes_path(col_item, "target.aspects.virtualDisksAspect.virtualDisks[0].businessAccount.vendorIds")
+      field "resourceGroup", jmes_path(col_item, "target.aspects.cloudAspect.resourceGroup.displayName")
+      field "resourceID", jmes_path(col_item, "target.vendorIds")
+      field "resourceName", jmes_path(col_item, "target.displayName")
+      field "region", jmes_path(col_item, "target.aspects.virtualDisksAspect.virtualDisks[0].dataCenter.displayName")
+      field "resourceType", jmes_path(col_item, "target.aspects.virtualDisksAspect.virtualDisks[0].tier")
+      field "provider", jmes_path(col_item, "target.aspects.virtualDisksAspect.virtualDisks[0].businessAccount.discoveredBy.type")
+      field "details", jmes_path(col_item, "details")
+      field "tags", jmes_path(col_item, "target.tags")
+      field "createdTime", jmes_path(col_item, "createTime")
+      field "attachmentState", jmes_path(col_item, "target.aspects.virtualDisksAspect.virtualDisks[0].attachmentState")
+      field "daysUnattached", jmes_path(col_item, "target.aspects.virtualDisksAspect.virtualDisks[0].numDaysUnattached")
+      field "size", jmes_path(col_item, "target.aspects.virtualDisksAspect.virtualDisks[0].stats")
+      field "savings", jmes_path(col_item, "stats[0].value")
+      field "savingsCurrency", jmes_path(col_item, "stats[0].units")
+    end
+  end
+end
+
+##this will potentially be a lot of calls. how to make this more performant
+datasource "ds_get_business_units" do
+  request do
+    run_script $js_get_business_units, $auth_cookie, $param_provider
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "uuid", jmes_path(col_item, "uuid")
+      field "accountID", jmes_path(col_item, "accountId")
+      field "displayName", jmes_path(col_item, "displayName")
+    end
+  end
+end
+
+datasource "ds_filtered_turbonomics_recommendations" do
+  run_script $js_filtered_turbonomics_recommendations, $ds_get_turbonomics_recommendations, $param_days_unattached, $ds_get_business_units, $param_provider
+end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+##script using a manually acquired turbonomic cookie and pulls hard coded unused volume data
+script "js_get_turbonomics_recommendations", type: "javascript" do
+  result "request"
+  parameters "auth_cookie"
+  code <<-EOS
+  //replace cookie every day this is run
+    var request = {
+      verb: "POST",
+      pagination: "turbonomics_volumes_pagination_header",
+      host: "sales1.demo.turbonomic.com",
```

Comment:
You should manage this value as param, otherwise you're forcing user to always use this host and if he got cookie from another one, he'll get an error, it applies for all cases where you set `sales1.demo.turbonomic.com` hardcoded.